consensus/reactor.go

```diff
@@ -1080,7 +1080,7 @@ func (ps *PeerState) SetHasProposalBlockPart(height int64, round int32, index in
 func (ps *PeerState) PickSendVote(votes types.VoteSetReader) bool {
 	if vote, ok := ps.PickVoteToSend(votes); ok {
 		msg := &VoteMessage{vote}
-		ps.logger.Debug("Sending vote message", "ps", ps, "vote", vote)
+		ps.logger.Debug("Sending vote message", "vote", vote)
```

Comment:
I think it may return again in our future maintenance or upstream merge operation. I know it's difficult, but could you add a test case to prevent a recurrence, or at least keep the information in the source?

Reply:
Alternatively, this PR can be labeled emerge and modified later.

Reply:
> could you add a test case

Do I add `TestPickSendVote` in `consensus/reactor_test.go`? And should it inside check the debug logging?

> at least keep the information in the source?

I'll do that.

Reply:
As for testing, it is if you know a possible way.

If we could implement a logger for testing and check that the passed parameters don't contain an unlocked `PeerState`, we should be able to detect all of these problems with that logger.

But that seems a bit over the top, so I think just a comment could be fine.